### PR TITLE
bt-716: connect uploading photo and company logo

### LINKED ISSUE #716: Connect uploading photo and company logo
When talent/employer adds a profile photo/company logo it should be displayed on the Preview page and Profile page

### PR DESCRIPTION
**Pull request by illia-dobro**

### CODE REVIEW

File: mobile/src/bundles/common/store/actions.ts [hunk 1]
```diff
@@ -37,11 +64,52 @@ const updateOnboardingData = createAsyncThunk<
     UserDetailsGeneralRequestDto,
     AsyncThunkConfig
 >(`${sliceName}/updateOnboardingData`, async (stepPayload, { extra }) => {
-    const { commonApi, notifications } = extra;
+    const { commonApi, notifications, fileUploadApi } = extra;
     const { badges, hardSkills, photo, cv, companyLogo, ...payload } =
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/store/actions.ts]:
```suggestion
    const { badges, hardSkills, photo, cv: cvDocument, companyLogo, ...payload } =
```

File: mobile/src/bundles/common/store/actions.ts [hunk 2]
```diff
@@ -37,11 +64,52 @@ const updateOnboardingData = createAsyncThunk<
...
         stepPayload;
     const talentHardSkills = hardSkills?.map((skill) => skill.value);
 
+    const cvDocument = cv;
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/store/actions.ts]:
```suggestion
```

File: mobile/src/bundles/common/store/actions.ts [hunk 3]
```diff
@@ -50,14 +118,14 @@ const updateOnboardingData = createAsyncThunk<
...
         });
+
         return {
             ...response,
             //TODO remove when it is ready at the backend
             ...(hardSkills && { hardSkills }),
             ...(badges && { badges }),
             ...(photo && { photo }),
             ...(cv && { cv }),
-            ...(companyLogo && { companyLogo }),
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/store/actions.ts]: I think cv and photo also can be moved (please doublecheck)

File: package-lock.json
Comment (katerynakostikova) [inline package-lock.json:1]: if this autogenerated then ok, if not please remove this changes

**Comment by katerynakostikova** [comment]
Fix linter error

File: mobile/src/bundles/common/store/actions.ts [hunk 4]
```diff
@@ -7,16 +7,49 @@ import {
...
+            file: photo,
+        });
+    }
+
+    if (files.length > EMPTY_FILE_COUNT) {
+        const response = await fileUploadApi.upload({ files });
+        mapFilesToPayload({
+            payload: payload,
+            files: response,
+        });
```
Comment (katerynakostikova) [inline mobile/src/bundles/common/store/actions.ts]: do you use somewhere result of this function?

Comment (illia-dobro) [reply mobile/src/bundles/common/store/actions.ts]: It is return payload (updating)
payload

Comment (katerynakostikova) [reply mobile/src/bundles/common/store/actions.ts]: It should return the payload, do not modify the passed one! It's bad practice

**Review by katerynakostikova — APPROVED** [review]

**Comment by katerynakostikova** [comment]
Conflicts

**Review by likeri29 — APPROVED** [review]

**Pull request merged**